### DISCUSSION TITLE
How to run scvelo on cellbender background-removed results?

### DISCUSSION THREAD

**Discussion by sooheelee** [theislab/scvelo#1036] [Q&A]
I have a [cellbender](https://github.com/broadinstitute/CellBender) background-removed counts matrix. My understanding is velocyto runs on the cellranger BAM to get intronic and exonic read counts. I would like to know what is the correct approach to using cellbender filtered results in scvelo, e.g. does the loom file still apply? 

Part of answering this question is understanding how does scvelo uses the information from the counts matrix versus from the loom? I can confirm scvelo does use the counts from the counts matrix in that my scvelo results change when I substitute the matrix and hold everything else equal. 

Thank you in advance for discussion.

---
### Using original cellranger counts (first) versus cellbender background-removed counts (second)
clusters are the same
![image](https://user-images.githubusercontent.com/11543866/230970119-b125c0ca-b523-483c-8160-e5921b11c763.png)
![image](https://user-images.githubusercontent.com/11543866/230970139-d7a4a8cf-a581-4920-b31b-c84799e2786b.png)

color="latent_time"
![image](https://user-images.githubusercontent.com/11543866/230969323-e512bd2e-5b39-4a8c-a9c0-2ed6f1511af2.png)
![image](https://user-images.githubusercontent.com/11543866/230969439-466be84f-c534-463e-b7c7-a7b4ace8eb32.png)

same genes, sortby='latent_time'
![image](https://user-images.githubusercontent.com/11543866/230969343-ef767d91-c227-4057-9ca1-4f71ed0a9ffd.png)
![image](https://user-images.githubusercontent.com/11543866/230969455-a9141d42-7c08-4084-83ab-49168d8ba268.png)

here order on x forced
![image](https://user-images.githubusercontent.com/11543866/230969365-bdf2dd49-efdc-496d-b0d0-db89a43ff0e2.png)
![image](https://user-images.githubusercontent.com/11543866/230969472-0ece2c84-55a2-40e7-a7d9-413c179e3c84.png)

**Comment by Laura19993** [comment]
I am also interested in this topic.
What is the correct approach to using cellbender filtered results in scvelo?

**Comment by yulchen810** [comment]
> I have a [cellbender](https://github.com/broadinstitute/CellBender) background-removed counts matrix. My understanding is velocyto runs on the cellranger BAM to get intronic and exonic read counts. I would like to know what is the correct approach to using cellbender filtered results in scvelo, e.g. does the loom file still apply? 
> 
> Part of answering this question is understanding how does scvelo uses the information from the counts matrix versus from the loom? I can confirm scvelo does use the counts from the counts matrix in that my scvelo results change when I substitute the matrix and hold everything else equal. 
> 
> Thank you in advance for discussion.
> 
> ---
> ### Using original cellranger counts (first) versus cellbender background-removed counts (second)
> clusters are the same
> ![image](https://user-images.githubusercontent.com/11543866/230970119-b125c0ca-b523-483c-8160-e5921b11c763.png)
> ![image](https://user-images.githubusercontent.com/11543866/230970139-d7a4a8cf-a581-4920-b31b-c84799e2786b.png)
> 
> color="latent_time"
> ![image](https://user-images.githubusercontent.com/11543866/230969323-e512bd2e-5b39-4a8c-a9c0-2ed6f1511af2.png)
> ![image](https://user-images.githubusercontent.com/11543866/230969439-466be84f-c534-463e-b7c7-a7b4ace8eb32.png)
> 
> same genes, sortby='latent_time'
> ![image](https://user-images.githubusercontent.com/11543866/230969343-ef767d91-c227-4057-9ca1-4f71ed0a9ffd.png)
> ![image](https://user-images.githubusercontent.com/11543866/230969455-a9141d42-7c08-4084-83ab-49168d8ba268.png)
> 
> here order on x forced
> ![image](https://user-images.githubusercontent.com/11543866/230969365-bdf2dd49-efdc-496d-b0d0-db89a43ff0e2.png)
> ![image](https://user-images.githubusercontent.com/11543866/230969472-0ece2c84-55a2-40e7-a7d9-413c179e3c84.png)
> 
> 

Hi, my 10x data also processed by cellbender  using cellranger outputs. some cells are removed during cellbender step.
should I load cellbender outputs to velocyto to produce loom file?
